libr/socket/run.c

```diff
@@ -475,6 +475,100 @@ R_API const char *r_run_help() {
 	"# nice=5\n";
 }
 
+static int fd_forward (int in_fd, int out_fd, char **buff) {
+	int size = 0;
+
+	if (ioctl (in_fd, FIONREAD, &size) == -1) {
+		perror ("ioctl");
+		return -1;
+	}
+
+	if (size == 0) // child process exited or socket is closed
+		return -1;
+
+	*buff = realloc (*buff, size);
+	read (in_fd, *buff, size);
+
+	if (write (out_fd, *buff, size) != size) {
+		perror ("write");
+		return -1;
+	}
+	return 0;
+}
+
+static int redirect_socket_to_pty (RSocket *sock) {
+#if __UNIX__ && !__ANDROID__ && LIBC_HAVE_FORK
```

Comment:
not sure how many other unixes restrict access to PTY. better use #if HAVE_PTY and define this variable at the begining of the file